src/index.ts

```diff
@@ -36,6 +36,40 @@ async function redirectWithFlash(url: string, flash: FlashSessionValues, init?:
   });
 }
 
+/**
+ * Helper method used to display a success toast notification without redirection
+ *
+ * It need to be awaited
+ * @param successMessage Message to be shown as success
+ * @returns Returns successMessage and a toast cookie set
+ */
+export async function displaySuccessMessage(successMessage: string) {
```

Comment:
I would not do it like this, I would accept an object generic as the first argument and the toast: toastMessage as the second, and init?: ResponseInit as the third so the signature is identical to the normal ones, and then return:
```jsx
return json(
    { ...passedInData, toast },
    {
      headers: await flashMessage({
        toast: { message: successMessage, type: "success" },
      }),
    },
```
Also I would call it `jsonWithSuccess`